near-sdk/src/types/error.rs

```diff
@@ -32,3 +32,30 @@ where
         crate::env::panic_str(self.as_ref())
     }
 }
+
+/// A simple type used in conjunction with [FunctionError] representing that the function should
+/// abort without a custom message.
+///
+/// ```
+/// use near_sdk::borsh::{BorshDeserialize, BorshSerialize};
+/// use near_sdk::{Abort, near_bindgen};
+///
+/// #[near_bindgen]
+/// #[derive(Default, BorshDeserialize, BorshSerialize)]
+/// pub struct Contract;
+///
+/// #[near_bindgen]
+/// impl Contract {
+///     #[handle_result]
+///     pub fn foo(&self) -> Result<Result<String, &'static str>, Abort> {
+///         Ok(Ok("success".to_string()))
+///     }
+/// }
+/// ```
+pub struct Abort;
```

Comment:
prob best to derive Debug, Clone, PartialEq to not be prickly to use in tests and otherwise

Reply:
maybe also `impl std::error::Error` just in case someone wants to return `Box<dyn Error>` for multiple error types in the same function

Reply:
I disagree with this one. The point of this is to be optimized for minimal code size, so having this would make it silently use the formatting machinery for errors.

A bit opinionated so do either, but I think we only add this if there are valid requests from the community. I can't think of when this would be useful

Reply:
Yeah, I would rather leave out `std::error::Error` implementation. If the need arises, we can always add it.